stdlib/builtins.pyi

```diff
@@ -156,6 +157,8 @@ class type(object):
     def __subclasscheck__(self, subclass: type) -> bool: ...
     @classmethod
     def __prepare__(metacls, __name: str, __bases: Tuple[type, ...], **kwds: Any) -> Mapping[str, Any]: ...
+    if sys.version_info >= (3, 10):
+        def __or__(self, t: Any) -> types.Union: ...
```

Comment:
```suggestion
        def __or__(self, t: Any) -> Any: ...
```
As you mentioned, `int | int` returns `int`, not a `Union`. We could do something more clever with overloads, but I'm not sure that's worth it since type checkers will surely special-case this.

Reply:
The type of ```int | int``` will still be ```types.Union```
![image](https://user-images.githubusercontent.com/20560572/113988525-bf20e180-9879-11eb-9a3c-3826b82466b1.png) so we can leave it as it is even for such cases.

Reply:
Ah, I was confused because it gets printed as `int`, but it's indeed a Union object at runtime. Sorry for the confusion.

```
Jelles-Air:cpython jelle$ ./python.exe 
Python 3.10.0a7+ (heads/master:28d28e053d, Apr  8 2021, 07:40:29) [Clang 11.0.3 (clang-1103.0.32.59)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> int | int
int
>>> type(_)
<class 'types.Union'>
```